### PR TITLE
feat(rnd): Add initial block execution credit accounting UI on AutoGPT Builder

### PR DESCRIPTION
**Pull request by majdyz**
### Background

Credit accounting is required to limit the excessive execution of costly blocks.

### Changes 🏗️

Added initial UI for block execution credit accounting.

https://github.com/user-attachments/assets/38585689-2d97-479d-bb73-de0a2cfcffce

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

**Review by Swiftyos — DISMISSED** [review]

**Comment by majdyz** [comment]
@aarushik93 would you mind reviewing this bit: https://github.com/Significant-Gravitas/AutoGPT/pull/8078/commits/f73d52979961fd01f9bfdc06c211dbb10b4cbfa9

This PR has been pre-approved but I need to include this change to make it run work with the latest master change on local supabase.

**Review by aarushik93 — APPROVED** [review]

File: rnd/autogpt_server/autogpt_server/util/settings.py [hunk 1]
```diff
@@ -95,6 +100,11 @@ class Config(UpdateTrackingModel["Config"], BaseSettings):
         description="The port for agent server daemon to run on",
     )
 
+    agent_api_port: int = Field(
+        default=8006,
+        description="The port for agent server API to run on",
+    )
+
```
Comment (Pwuts) [inline rnd/autogpt_server/autogpt_server/util/settings.py:107]: This seems unrelated, wasn't announced, isn't mentioned in the PR description, and just cost me half an hour of debugging.
